src/lib/hooks/useZoom.ts

```diff
@@ -27,24 +37,27 @@ export const useZoom = ({
             e.preventDefault();
             let i = 0;
             let newScale = scale;
-            const timer = setInterval(() => {
+            if (interval.current) {
+                clearInterval(interval.current);
+            }
+            interval.current = setInterval(() => {
                 newScale = newScale + (quad(-4 + i++) / 4) * delta;
                 setScale(Math.min(maxScale, Math.max(minScale, newScale)));
-                if (i > 8) {
-                    clearInterval(timer);
+                if (i > 8 && interval.current) {
```

Comment:
Er dette en ekstra sjekk for om zoom skal stoppe? Dvs. det som skjer først av 50 millisekunder eller 8 "zooms" skal gjøre at dobbelklikk er ferdig?

Her spør jeg generelt om koden og ikke endringen spesifikt.

Reply:
This is an interval that is called every 50 ms and is stopped after it has been called eight times. The purpose is to create a smooth zoom.

Reply:
makes sense!